File: test/entity/area.ts
import { Entity, Id, Column, ManyToOne, JoinColumn, OneToMany } from '../../core/decorator/decorator'
import { BaseEntity } from '../../core/baseentity'
import { EFkConstraint } from '../../core/entitydefine'
import {Agent} from './agent'
import {Provider} from './provider'
import {Road} from './road'

@Entity("t_area",'tunnel')
export class Area extends BaseEntity{
	@Id()
	@Column({
		name:'area_id',
		type:'int',
		nullable:false
	})
	private areaId:number;

	@ManyToOne({entity:'Area',eager:false})
	@JoinColumn({name:'parent_id',refName:'area_id'})
	private area:Area;

	@Column({
		name:'area_name',
		type:'string',
		nullable:true
	})
	private areaName:string;

	@Column({
		name:'area_code',
		type:'string',
		nullable:true
	})
	private areaCode:string;

	@Column({
		name:'zip_code',
		type:'string',
		nullable:true
	})
	private zipCode:string;

	@Column({
		name:'depth',
		type:'int',
		nullable:true
	})
	private depth:number;

	@OneToMany({entity:'Agent',onDelete:EFkConstraint.SETNULL,onUpdate:EFkConstraint.CASCADE,mappedBy:'area',eager:false})
	private agents:Array<Agent>;

	@OneToMany({entity:'Area',onDelete:EFkConstraint.SETNULL,onUpdate:EFkConstraint.CASCADE,mappedBy:'area',eager:false})
	private areas:Array<Area>;

	@OneToMany({entity:'Provider',onDelete:EFkConstraint.SETNULL,onUpdate:EFkConstraint.CASCADE,mappedBy:'area',eager:false})
	private providers:Array<Provider>;

	@OneToMany({entity:'Road',onDelete:EFkConstraint.SETNULL,onUpdate:EFkConstraint.CASCADE,mappedBy:'area',eager:false})
	private roads:Array<Road>;

	public getAreaId():number{
		return this.areaId;
	}
	public setAreaId(value:number){
		this.areaId = value;
	}

	public getArea():Area{
		return this.area;
	}
	public setArea(value:Area){
		this.area = value;
	}

	public getAreaName():string{
		return this.areaName;
	}
	public setAreaName(value:string){
		this.areaName = value;
	}

	public getAreaCode():string{
		return this.areaCode;
	}
	public setAreaCode(value:string){
		this.areaCode = value;
	}

	public getZipCode():string{
		return this.zipCode;
	}
	public setZipCode(value:string){
		this.zipCode = value;
	}

	public getDepth():number{
		return this.depth;
	}
	public setDepth(value:number){
		this.depth = value;
	}

	public getAgents():Array<Agent>{
		return this.agents;
	}
	public setAgents(value:Array<Agent>){
		this.agents = value;
	}

	public getAreas():Array<Area>{
		return this.areas;
	}
	public setAreas(value:Array<Area>){
		this.areas = value;
	}

	public getProviders():Array<Provider>{
		return this.providers;
	}
	public setProviders(value:Array<Provider>){
		this.providers = value;
	}

	public getRoads():Array<Road>{
		return this.roads;
	}
	public setRoads(value:Array<Road>){
		this.roads = value;
	}

}

File: jpa/pojo/Area.java
package dao.pojo;
// default package

import java.util.HashSet;
import java.util.Set;
import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToMany;
import javax.persistence.Table;
import javax.persistence.TableGenerator;


/**
 * Area entity. @author MyEclipse Persistence Tools
 */
@Entity
@Table(name="t_area"
    ,catalog="tiandian"
)

public class Area extends BasePojo  implements java.io.Serializable {


    // Fields    

     private Long area_id;
     private Area area;
     private String area_name;
     private String areacode;
     private String zipcode;
     private Integer depth;
     private Integer is_open;		//是否开通
     private Set<Area> areas = new HashSet<Area>(0);
    // Constructors

    /** default constructor */
    public Area() {
    }

	/** minimal constructor */
    public Area(Long area_id) {
        this.area_id = area_id;
    }
    
    /** full constructor */
    public Area(Long area_id, Area area, String area_name, String areacode, String zipcode, Integer depth, Set<Area> areas) {
        this.area_id = area_id;
        this.area = area;
        this.area_name = area_name;
        this.areacode = areacode;
        this.zipcode = zipcode;
        this.depth = depth;
        this.areas = areas;
    }

   
    // Property accessors
    @Id 
    @TableGenerator(name = "AREA_GEN", table = "sequence", pkColumnName = "SEQ_NAME", valueColumnName = "SEQ_COUNT", pkColumnValue = "SEQ_AREA", allocationSize = 1)
	@GeneratedValue(strategy = GenerationType.TABLE, generator = "AREA_GEN")
    @Column(name="area_id", unique=true, nullable=false)
    public Long getArea_id() {
        return this.area_id;
    }
    
    public void setArea_id(Long area_id) {
        this.area_id = area_id;
    }
	@ManyToOne(fetch=FetchType.LAZY)
    @JoinColumn(name="parent_id")

    public Area getArea() {
        return this.area;
    }
    
    public void setArea(Area area) {
        this.area = area;
    }
    
    @Column(name="area_name", length=50)

    public String getArea_name() {
        return this.area_name;
    }
    
    public void setArea_name(String area_name) {
        this.area_name = area_name;
    }
    
    @Column(name="areacode", length=4)

    public String getAreacode() {
        return this.areacode;
    }
    
    public void setAreacode(String areacode) {
        this.areacode = areacode;
    }
    
    @Column(name="zipcode", length=6)

    public String getZipcode() {
        return this.zipcode;
    }
    
    public void setZipcode(String zipcode) {
        this.zipcode = zipcode;
    }
    
    @Column(name="depth")

    public Integer getDepth() {
        return this.depth;
    }
    
    public void setDepth(Integer depth) {
        this.depth = depth;
    }
    
    
    @Column(name="is_open")
    public Integer getIs_open() {
		return is_open;
	}

	public void setIs_open(Integer is_open) {
		this.is_open = is_open;
	}

	@OneToMany(cascade=CascadeType.ALL, fetch=FetchType.LAZY, mappedBy="area")
    public Set<Area> getAreas() {
        return this.areas;
    }
    
    public void setAreas(Set<Area> areas) {
        this.areas = areas;
    }
   
	@Override
	public Object getEntityId() {
		return this.area_id;
	}

}